Lib/asyncio/events.py

```diff
@@ -101,7 +101,6 @@ class TimerHandle(Handle):
     __slots__ = ['_scheduled', '_when']
 
     def __init__(self, when, callback, args, loop, context=None):
-        assert when is not None
```

Comment:
Why did you delete this check?

Reply:
`loop.call_at(...)` is the only place where TimeHandler created, call_at() has a check now.

Do you prefer to check twice?

Reply:
Is TimeHandle not a public API?

Reply:
Good question, I have no strong answer for it.
From my perspective, Handle, TimerHandle, Future, Task, and transports are part of public API but their constructors are not. All these objects should be instantiated by factory methods only.
Also, all these constructors are not documented.

Currently, asyncio checks `coro` argument in `Task.__init__` but blindly trusts all arguments in transport constructors.
I believe, the task case is a result of a very old and discouraged design when we used to allow `task = Task(async_func())` usage. Now it is soft-deprecated without raising a warning.

Regarding `TimerHandle`: the direct `th = asyncio.TimerHandle(...)` call doesn't make any sense; there is no public way to register it in the loop. An alone handle is simply useless.

Reply:
Okay.